reTurn/AsyncUdpSocketBase.cxx

```diff
@@ -67,17 +67,27 @@ AsyncUdpSocketBase::connect(const std::string& address, unsigned short port)
 
 void 
 AsyncUdpSocketBase::handleUdpResolve(const asio::error_code& ec,
-                                     asio::ip::udp::resolver::iterator endpoint_iterator)
+                                     asio::ip::udp::resolver::results_type results)
 {
    if (!ec)
    {
-      // Use the first endpoint in the list. 
-      // Nothing to do for UDP except store the connected address/port
+      // Find the first endpoint in the list matching the local endpoint protocol type
+      asio::ip::udp::resolver::results_type::const_iterator       it  = results.begin();
+      const asio::ip::udp::resolver::results_type::const_iterator end = results.end();
+      for (; it != end; ++it)
+      {
+         const asio::ip::udp::resolver::results_type::endpoint_type &ep = it->endpoint();
+         if (ep.protocol() == mSocket.local_endpoint().protocol())
+         {
       mConnected = true;
```

Comment:
Some things look misaligned here.  It's hard to tell if it's just a github display issue.  Please make sure you are using spaces and not TABs for all resip changes, as this is the current coding convention.  see https://github.com/resiprocate/resiprocate/wiki/Code-Conventions